validators/dns.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+from .utils import validator
+
+
+pattern = re.compile(
+    r'^((?![0-9]{1,63}[.])(?![0-9]+$)'
+    r'(?!-)([a-zA-Z0-9-_]{0,63}(?<!-)([.])?)){1,32}$'
+)
+
+
+@validator
+def label(value):
```

Comment:
Would hostname made more sense as a name for this? Label sounds a bit too generic for me